### ISSUE TITLE
error: Filename patterns error if it has square brackets in it

### ISSUE THREAD

**Issue by rofrol** [nushell/nushell#933]
**Describe the bug**

- No autoescape special chars like `[` or `]` when pressing `Tab`.
- Even when added escapes manually, does not work.
- Cannot rename such file even when wrapping with `"`.
- Can only move to some directory.

**To Reproduce**
1. `touch file[asdf].mkv`
2. `mkdir t`
3. `mv file[asdf].mkv t/`

this does not work

`mv file\[asdf\].mkv t/`

this works

`mv "file[asdf].mkv" t/`

this does not work

`mv "file[asdf].mkv file2.mkv`

**Expected behavior**

nu should autoescape special chars like bash does it so after pressing `Tab` it completes file name to `file\[asdf\].mkv` and no errors when using `mv` with such files.

**Screenshots**

**Configuration (please complete the following information):**
 - OS: Ubuntu 19.10
 - Version 0.5.0
- `nu` set as default shell in gnome-terminal as described here https://askubuntu.com/questions/1180222/how-to-set-zsh-as-default-for-gnome-terminal-only/1180251#1180251

**Comment by thegedge** [comment]
Hey, @rofrol. For a little background, we've been leaning towards not having a backslash escape character, if we can avoid it, to encourage explicitness in arguments (from quoting).

That being said, in this case we should at least still wrap things in double quotes when completing. In other words, something like this:

    ❯ mv file[as<TAB>

becomes

    ❯ mv "file[asdf].mkv"

Would that also work for you?

**Comment by yanganto** [comment]
I got a similar issue with round brackets

OS: Archlinux
Version: 0.16.0

```
/tmp/> unzip "round-0-2(2).zip"
sh: -c: line 0: syntax error near unexpected token `('
sh: -c: line 0: `unzip round-0-2(2).zip'
```

**Comment by thegedge** [comment]
I believe 0.17.0 will fix this issue.

**Comment by rofrol** [comment]
Not fixed in 0.34
